docs/content/docs/osm_config_map.md

```diff
@@ -4,24 +4,141 @@ description: "OSM ConfigMap"
+| tracing_endpoint | string | /api/v2/spans | `kubectl patch ConfigMap osm-config -n osm-system -p '{"data":{"tracing_endpoint":"/abracadabra"}}' --type=merge` |
+| tracing_port| int | `"9411"` | `kubectl patch ConfigMap osm-config -n osm-system -p '{"data":{"tracing_port":"1234"}}' --type=merge` |
+
+## Validating Webhook
+
+The validating webhook validates changes made to the default fields in the `osm-config` . It prevents users who are updating configurable field values with `kubectl` commands from using values beyond the tested limits. Default fields of the osm-config are the keys in the table [above](#ConfigMap-Values) and can also be found in the [`osm-config` yaml](https://github.com/openservicemesh/osm/blob/main/charts/osm/templates/osm-configmap.yaml).
+
+The validating webhook yaml can be found [here](https://github.com/openservicemesh/osm/blob/main/charts/osm/templates/validatingwebhook.yaml) and the validation code [here](https://github.com/openservicemesh/osm/blob/main/pkg/configurator/validating_webhook.go). The validating webhook will accept or reject the changed ConfigMap configuration for the value(s) based on the [allowed values](#ConfigMap-Values) for that field(s).
+
+To get details on the validating webhook resource use `kubectl describe` command.
+```bash
+#  The validating webhook name is set as `osm-webhook-<mesh_name>`
+kubectl describe validatingwebhookconfiguration osm-webhook-osm
+```
+### Validating Webhook Reason for Denials
+
+| Fields | Reasons for Denial |
+|--------|--------------------|
+| egress | `must be a boolean` |
+| enable_debug_server | `must be a boolean` |
+| enable_privileged_init_container| `must be a boolean` |
+| envoy_log_level | `invalid log level` |
+| outbound_ip_range_exclusion_list | `must be a list of valid IP addresses of the form a.b.c.d/x` |
+| permissive_traffic_policy_mode | `must be a boolean` |
+| prometheus_scraping | `must be a boolean` |
+| service_cert_validity_duration | `invalid time format must be a sequence of decimal numbers each with optional fraction and a unit suffix` |
+| tracing_enable | `must be a boolean` |
+| tracing_port| <ul><li>`must be an integer`</li><li>`must be between 0 and 65535`</li></ul> |
+| use_https_ingress | `must be a boolean` |
+
+> Any changes to the OSM ConfigMap metadata will be rejected with `cannot change metadata`.
+
+### List of Default Fields in ConfigMap
+- egress
+- enable_debug_server
+- enable_privileged_init_container
+- envoy_log_level
+- permissive_traffic_policy_mode
+- prometheus_scraping
+- service_cert_validity_duration
+- tracing_enable
+- use_https_ingress
+
+> All default fields must be present in `osm-config`, removal of one or more default fields will error with `<field>: must be included as it is a default field`.
+
+# Troubleshooting Guide
+See [here](docs/content/docs/troubleshooting/osm_config_map.md) for ConfigMap troubleshooting.
```

Comment:
Are we using relative links or absolute links or docs site links?

Reply:
let's say docs site links for now